serf/serf.go

```diff
@@ -1884,3 +1890,20 @@ func (s *Serf) NumNodes() (numNodes int) {
 
 	return numNodes
 }
+
+// ValidateNodeNames checks the ValidateNodeNames flag to be true
+// then runs validations
+func (s *Serf) ValidateNodeNames() error {
+	if s.config.ValidateNodeNames {
+		var InvalidNameRe = regexp.MustCompile(`[^A-Za-z0-9\-\.\\]+`)
+		if InvalidNameRe.MatchString(s.config.NodeName) {
+			return fmt.Errorf("NodeName contains invalid characters %v , Valid characters include "+
+				"all alpha-numerics and dashes.", s.config.NodeName)
```

Comment:
We should specify alphanumerics, dashes and `.`s. Same with the comments for `ValidateNodeNames` above.